src/station_type.h

```diff
@@ -22,7 +22,8 @@ struct RoadStop;
 struct StationSpec;
 struct Waypoint;
 
-static const StationID NEW_STATION = 0xFFFE;
+static const StationID NEW_STATION = 0xFFFD;
+static const StationID ADJACENT_STATION = 0xFFFE;
 static const StationID INVALID_STATION = 0xFFFF;
 
 typedef SmallStack<StationID, StationID, INVALID_STATION, 8, 0xFFFD> StationIDStack;
```

Comment:
Maybe
```suggestion
typedef SmallStack<StationID, StationID, INVALID_STATION, 8, 0xFFFC> StationIDStack;
```

Reply:
You'd think so, right? But it's the maximum number of elements, not the index of the last element. So you could consider that it used to be one too big in the past.

Although I'm wondering why it needed to be bigger than the station pool's size. Maybe it allows duplicates of something, I haven't looked into that (yet). In any case I see no reason why this constant needs to be changed.